File: Pramp/MoveZerosToEnd.js
// Given a static-sized array of integers arr, move all zeroes in the array to the end of the array. You should preserve the relative order of items in the array.

function moveZerosToEnd(array){
    let write = 0;
    for (let read = 0; i < array.length; read++){
        if (array[read] != 0){
            [array[write], array[read]] = [array[read], array[write]];
            write++;
        }
    }
    return array;
}

File: Pramp/MoveZerosToEnd.py
# Given a static-sized array of integers arr, move all zeroes in the array to the end of the array. You should preserve the relative order of items in the array.

def moveZerosToEnd(array):
    write = 0
    for read in range(len(array)):
        if array[read] != 0:
            array[read], array[write] = array[write], array[read]
            write += 1

    return array